src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -344,7 +345,7 @@ def reconstruct_messages(
                     )
                 continue
 
-            new_payload_value["retention_days"] = 90
+            new_payload_value["retention_days"] = quotas.get_event_retention(organization=org_id)
```

Comment:
Are the quotas being fetched from postgres? I am wondering whether there would be any perf impact on performing these lookups for each message. It might not be a problem if these values are cached. 

This is not a blocking comment. I am open for testing it out in production and reverting if we find issues

Reply:
postgres is the ultimate source of truth (https://github.com/getsentry/getsentry/blob/ed65584f86ed951cf3fcabfbdad766a3e73e3baf/getsentry/quotas.py#L449-L455) but I'm not 100% sure how caching works.

Given that the eventstream flow uses this method, I'm guessing that org subscriptions are cached.